ddl/column.go

```diff
@@ -796,6 +790,12 @@ func (w *worker) doModifyColumnTypeWithData(
 			}
 		})
 		ver, err = updateVersionAndTableInfoWithCheck(t, job, tblInfo, originalState != changingCol.State)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+		// Make sure job args change behind the `updateVersionAndTableInfoWithCheck`, otherwise, the job args will
+		// be updated in `finishDDLJob` even if it meet a error in `updateVersionAndTableInfoWithCheck`.
```

Comment:
`finishDDLJob` -> `updateDDLJob`

Please add a test.

Reply:
good, I've listed the manual test script, does it still ...

Reply:
I think we can add a UT.